### PR TITLE
add billing project id support to firewall endpoint resource

### PR DESCRIPTION
**Pull request by maxi-cit**
Adds support for optional 'description' field and required 'billingProjectId' field to 'google_network_security_firewall_endpoint'

Part of: https://github.com/hashicorp/terraform-provider-google/issues/15779
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/17396

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
networksecurity: added required field `billing_project_id` to `google_network_security_firewall_endpoint` resource. Any configuration without `billing_project_id` specified will cause resource creation fail (beta)
```

### CODE REVIEW

File: mmv1/products/networksecurity/FirewallEndpoint.yaml
Comment (shuyama1) [inline mmv1/products/networksecurity/FirewallEndpoint.yaml:118]: Will not providing the billing project make the resource creation failed in all cases?
Adding a required field is a breaking change that we'd wait until the next major release, unless without the field specified, all current configurations will fail. See details at https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes/#field-level-breaking-changes and https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/make-a-breaking-change/#in-minor-releases

File: mmv1/products/networksecurity/FirewallEndpoint.yaml
Comment (shuyama1) [inline mmv1/products/networksecurity/FirewallEndpoint.yaml:88]: would you mind adding this field in a test?

**Review by shuyama1 — CHANGES_REQUESTED** [review]

File: mmv1/products/networksecurity/FirewallEndpoint.yaml
Comment (maxi-cit) [reply mmv1/products/networksecurity/FirewallEndpoint.yaml:118]: Hey @shuyama1, thanks for the quick feedback. This breaks all tests cases according to [this](https://github.com/hashicorp/terraform-provider-google/issues/17396).
Unfortunately, this field was introduced as a required field after the enhacement PR was merged.

Comment (shuyama1) [reply mmv1/products/networksecurity/FirewallEndpoint.yaml:118]: Would you mind updating the test to include this field?

Comment (maxi-cit) [reply mmv1/products/networksecurity/FirewallEndpoint.yaml:114]: Hello @shuyama1 I added the field to the basic test so it fixes it.

File: mmv1/products/networksecurity/FirewallEndpoint.yaml
Comment (maxi-cit) [reply mmv1/products/networksecurity/FirewallEndpoint.yaml]: I removed this field from this PR because API is not saving this value and causes tests to fail

**Comment by maxi-cit** [comment]
Hello @shuyama1 is there anything else blocking this merge?

**Comment by shuyama1** [comment]
@maxi-cit  Thanks for the ping. The test skipped in VCR. I triggered a independent test run on [TC](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_NETWORKSECURITY/120767?buildTab=overview&focusLine=NaN&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildDeploymentsSection=false) and will update the result here when test's done (it may take ~20-30 mins).

Overall implementation LGTM. I do have one question regarding the test: Is there any particular reason that we need to use ADC  in this test?

**Comment by maxi-cit** [comment]
Hey @shuyama1, thanks for the update. I used ADC just for a matter of convinience so I could iterate on the test faster

**Comment by shuyama1** [comment]
> Hey @shuyama1, thanks for the update. I used ADC just for a matter of convinience so I could iterate on the test faster

Oh, got it! Then can we remove the provider override and what's modified in the CheckDestroy function? It'd be easier to read for future reference. I've tested it locally and they should be working fine with just `billing_project_id` specified in the resource in our testing environment. Thank you and apologize for the extra work.

**Comment by maxi-cit** [comment]
Hey @shuyama1 I removed the ADC logic could you run the tests, please?

**Review by shuyama1 — APPROVED** [review]
LGTM! Triggered skipped tests on [TC](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_NETWORKSECURITY/120895?buildTab=tests&expandedTest=build%3A%28id%3A120895%29%2Cid%3A2000000000) and tests passed!

Thank you!

**Pull request merged**
